### PR TITLE
completed unit-tests-jest exercise - Evan Hong

### PR DESCRIPTION
**Pull request by EvanS-Hong**
image

### CODE REVIEW

File: unit-tests-jest/numbers.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ */
+export function evenNumbers(numbers) {
+  return numbers.filter((n) => (n % 2 === 0));
+}
+
+/**
+ * Returns a number formatted in dollars and cents.
+ */
+export function toDollars(amount) {
+  return `$${amount}.00`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js]: This isn't the correct implementation. It only works for integers.

File: unit-tests-jest/numbers.test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    expect(result).toEqual([4, 10, 0]);
+  });
+});
+
+describe('toDolars', () => {
+  it('returns a number in $ format', () => {
+    const amount = 3;
+    const result = toDollars(amount);
+    expect(result).toEqual(`$${amount}.00`);
+  });
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:26]: Test a fractional number.

File: unit-tests-jest/numbers.test.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    const multiplier = 2;
+    const result = multiplyBy(obj, multiplier);
+    expect(result).toEqual({
+      name: 'jacob',
+      age: 20,
+      grade: 'A',
+      gpa: 4
+    });
+  });
+});
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js:61]: You only have happy path tests in this suite. You should test edge cases as well.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: unit-tests-jest/numbers.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ export function evenNumbers(numbers) {
  * Returns a number formatted in dollars and cents.
  */
 export function toDollars(amount) {
-  return `$${amount}.00`;
+  return `$${amount}`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js]: Still not right. It should format to dollars and cents. Try formatting `12.4335`

File: unit-tests-jest/numbers.test.js [hunk 5]
```diff
@@ -6,13 +6,23 @@ describe('evenNumbers', () => {
...
+    expect(result).toEqual([]);
+  });
 });
 
 describe('toDolars', () => {
   it('returns a number in $ format', () => {
-    const amount = 3;
+    const amount = 3.12;
+    const result = toDollars(amount);
+    expect(result).toEqual(`$${amount}`);
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js]: In a unit test, you want to hard-code the value you are testing against, to make certain the function is generating the correct value. So test against `$3.12`. Below, you need to decide what you want negative values to look like and hard-code that.

**Review by RobertGardner — CHANGES_REQUESTED** [review]

File: unit-tests-jest/numbers.js [hunk 6]
```diff
@@ -9,7 +9,10 @@ export function evenNumbers(numbers) {
  * Returns a number formatted in dollars and cents.
  */
 export function toDollars(amount) {
-  return `$${amount}`;
+  const splitValue = amount.toString().split('.');
+  const dollar = splitValue[0];
+  const cents = splitValue[1];
+  return `$${dollar}.${cents}`;
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.js]: Have you forgotten how we did this in an earlier exercise? Just use `.toFixed(2)`.

File: unit-tests-jest/numbers.test.js [hunk 7]
```diff
@@ -15,14 +15,14 @@ describe('evenNumbers', () => {
 
 describe('toDolars', () => {
   it('returns a number in $ format', () => {
-    const amount = 3.12;
+    const amount = 12.4335;
     const result = toDollars(amount);
-    expect(result).toEqual(`$${amount}`);
+    expect(result).toEqual('$12.4335');
```
Comment (RobertGardner) [inline unit-tests-jest/numbers.test.js]: No, it needs to round to just cents ('$12.43'). That's why we hard-code the values, so we can be certain we're getting what we want.

**Review by RobertGardner — CHANGES_REQUESTED** [review]
Sorry, one more round should do it, though!

**Review by RobertGardner — APPROVED** [review]
